pkg/adsc/delta.go

```diff
@@ -385,6 +430,102 @@ func initWatch(typeURL string, resourceName string) Option {
 	}
 }
 
+// DeltaMcpHandler is a function that will be called when a delta MCP config resource is received.
+// The entity is converted from a original `mcp.Resource`.
+type DeltaMcpConfigHandler func(gvk config.GroupVersionKind, namespaced, name string, entity *config.Config)
+
+func WithConfigStore(store model.ConfigStore) DeltaMcpConfigHandler {
+	return func(gvk config.GroupVersionKind, namespace, name string, cfg *config.Config) {
```

Comment:
For such complex function, please register it directly dutring delta client initiation. I donot see any advantages here